### DISCUSSION TITLE
How to query for Top level object and field in parallel

### DISCUSSION THREAD

**Discussion by coolFreight** [graphql-java/graphql-java#2659] [Q&A]
Hi I am trying to figure out is it possible to query for a top level object and one of its fields in parallel.   Here is an example

`
schema {
 query : GetData
}

type GetData {
     dataId (id :  String!) : DataObject
}

type DataObject {
   id:       String
   field1 : String
   field2: String
}
`

Let's say DataObject is persisted in the database with field1 data.  field2 data has to be retrieved from another source other than the database.

How can I load DataObject from the database and make a call for field2 data in parallel?

**Comment by bbakerman** [comment]
First have a read of https://www.graphql-java.com/blog/deep-dive-data-fetcher-results

This is the idea that in the  parent data fetcher you need to look ahead at the sub field selection and then  you can go and get data you need for the child field in the parent fetcher.

You can make this a parralel call in your parent data fetcher by using `CompleteableFuture` results

The code trick is you need to return a data object that the child fetcher (which will run) will be able to know how to get  its portion of the pre-fetched data.  This is either a custom designed object OR the use of local context to hold a custom object

**Comment by coolFreight** [comment]
Thanks. This helped me create a solution.
